### PR TITLE
[Merged by Bors] - feat: port GroupTheory.DoubleCoset

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
I'm unsure about the capitalization of some things:
 - I changed the namespace from `doset` to `Doset`
 - I changed `setoid` to `Setoid`
 Please double check if those were correct changes.

File: Mathlib/GroupTheory/DoubleCoset.lean [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+-- porting note: removed import
+-- import Mathlib.Tactic.Group
+
+variable {G : Type _} [Group G] {α : Type _} [Mul α] (J : Subgroup G) (g : G)
+
+namespace Doset
+
+open Pointwise
+
+/-- The double_coset as an element of `set α` corresponding to `s a t` -/
```
Comment (jcommelin) [inline Mathlib/GroupTheory/DoubleCoset.lean:39]:
```suggestion
/-- The double coset as an element of `set α` corresponding to `s a t` -/
```

File: Mathlib/GroupTheory/DoubleCoset.lean [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+theorem eq_of_not_disjoint {H K : Subgroup G} {a b : G}
+    (h : ¬Disjoint (doset a H K) (doset b H K)) : doset a H K = doset b H K := by
+  rw [disjoint_comm] at h
+  have ha : a ∈ doset b H K := mem_doset_of_not_disjoint h
+  apply doset_eq_of_mem ha
+#align doset.eq_of_not_disjoint Doset.eq_of_not_disjoint
+
+/-- The setoid defined by the double_coset relation -/
+def Setoid (H K : Set G) : Setoid G :=
```
Comment (jcommelin) [inline Mathlib/GroupTheory/DoubleCoset.lean:78]:
```suggestion
def setoid (H K : Set G) : Setoid G :=
```
This is a term, so it should start with a lower case letter.

**Review by jcommelin — COMMENTED** [review]
Minor comments, otherwise LGTM

bors d+

**Pull request closed without merging**